x/programs/rust/sdk-macros/src/lib.rs

```diff
@@ -90,7 +88,7 @@ pub fn public(_: TokenStream, item: TokenStream) -> TokenStream {
                     fn_arg.span(),
                     "Functions with the `#[public]` attribute can only ignore the first parameter.",
                 )),
-                _ => Ok(Ident::new(&format!("param_{i}"), fn_arg.span())),
+                _ => Ok(pat),
```

Comment:
🤔... we might want to put this back. I think a pattern match might break this:
```rust
fn add((a, b): (usize, usize)) -> usize {
    a + b
}
```
(this is a completely valid function signature)

Reply:
Feel free to back if you fix the struct construction.  I was struggling to get it to generate correctly.

Reply:
Yes and it is also going to be broken for destructured function parameters:

```rust
struct Hi {
    name: String,
    age: u64
}

fn third(one: u64, Hi { name, .. }: Hi) {}
```

For now, there could be a compile-time error that makes sure that the pattern of the type is unary.